### ISSUE TITLE
COG -> GeoTIFF() + normalize:false -> TileLayer() fails to draw on some Android devices, with render warning

### ISSUE THREAD

**Issue by potion-cellar** [openlayers/openlayers#15581]
**Describe the bug**
tl;dr On some small percentage of Android devices, GeoTIFF() with sources that have `normalize: false` fail to draw properly, sometimes on all browsers or sometimes only in a specific browser (like System WebView). When `normalize: true` is used they draw properly.

---
More details:

We display weather data using single band Float32 COGs that are added to an openlayers map by using a GeoTIFF source and a TileLayer.

For the sources within the GeoTIFF source, we use `normalize: false` and set a `min` and `max`. Then we'll typically do an interpolated or paletted style def with a colormap or whatever.

What we've seen consistently across hundreds of users for a variety of COGs is that some small percentage of Android devices (like 1-2%) will not draw the layer properly (usually it will be gray or completely transparent) -- sometimes across all browsers, sometimes only in a specific browser. This only happens if the sources use `normalize: false`. As soon as we change it to `normalize: true` the COGs start drawing properly again. We don't like to use `normalize: true` as it typically makes more logarithmic data unacceptably coarse near the minima.

This issue has not been reported on Windows, iOS, or MacOS.

The only pattern we've noticed across Android devices is that it has always been Google Pixels with the issue. But we've seen it reported across many versions of Android on many different Pixels, including the very latest Pixel and Android version, with no apparent pattern (other users with the exact same browser + pixel + android version will not have an issue). Though admittedly the information we've been able to get from our users has been a little lackluster.

On my Pixel 6 Pro, I am able to replicate the issue with the very latest version of Android System WebView (122.0.6261.64) on Android 14. This is output repeatedly to the console:

```
WebGL: INVALID_ENUM: texImage2D: invalid type
[.WebGL-0x204f60000]RENDER WARNING: texture bound to texture unit 0 is not renderable. It might be non-power-of-2 or have incompatible texture filtering (maybe)?
```

**To Reproduce**
1. Visit https://openlayers.org/en/latest/examples/cog-style.html ... maybe using Android 14 with Android System WebView 122.0.6261.64 -- though for me, earlier versions of Android and earlier versions of WebView also didn't work. but again, sort of only for my device (no issue on my friend's Pixel with same Android / WebView setup). 

2. If it doesn't play nice with your browser, you'll just see a black map. 

**Expected behavior**
I expect a TileLayer() to draw consistently whether or not it using a GeoTIFF source with `sources` that have `normalize: true` or `normalize: false`.

--

Please let me know if there's more information that would be useful for me to gather, I can probably shake some of users down a bit more or get more info out of WebView setup.

**Comment by mike-000** [comment]
See #13367 #13529 #14366 #14377 #15136

**Comment by potion-cellar** [comment]
Ah cool, didn't know a webgl2 port has been in the works for awhile. If more work is needed on a PR I'd be happy to jump in.

**Comment by potion-cellar** [comment]
As I've slowly wrapped my ahead around the issue and the context for everything I've rewritten the issue to be more straightforward hopefully. I also brought in some additional statistics in case it helps the `openlayers` maintainers make decisions about future WebGL2 implementation.

In the past few years `OES_texture_float` support has dropped from ~95% support on Android devices (at least from an archive.org version of webglstats.com in 2021) to ~88%, per the latest web3d survey data: https://web3dsurvey.com/webgl/extensions/OES_texture_float

According to [gpuinfo.org](https://opengles.gpuinfo.org/listreports.php?extension=GL_OES_texture_float&option=not), the following devices have limited (broken in some browsers) or no (broken in all browsers) `OES_texture_float` support:
 - Pixel 6, Pixel 6a
 - Pixel 7, Pixel 7 Pro
 - Pixel 8, Pixel 8 Pro
 - Samsung Galaxy A14 (B,P)
 - Samsung Galaxy A34 (E,M)
 - Samsung Galaxy A53 (B,E)
 - Samsung Galaxy S21 (SM-G991N, SM-G990E)
 - Oppo Reno8 Pro
 - Xiaomi 11T
 - Xiaomi Redmi 12, Note 12 Pro, 12C
 - Xiaomi 14T
 - Xiaomi Poco M5
 - Xiaomi Poco X6 Pro

Granted, this is not as problematic in the North America and Europe until the newer Pixel versions started gaining market share. For us, we have enough users on Pixel 7 & 8 where it is now a high priority production problem.
